extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQCompactionRunner.java

```diff
@@ -497,6 +502,7 @@ private Map<String, Object> createMSQTaskContext(CompactionTask compactionTask,
     context.put(QueryContexts.FINALIZE_KEY, false);
     // Only scalar or array-type dimensions are allowed as grouping keys.
     context.putIfAbsent(GroupByQueryConfig.CTX_KEY_ENABLE_MULTI_VALUE_UNNESTING, false);
+    context.putIfAbsent(MultiStageQueryContext.CTX_ARRAY_INGEST_MODE, "array");
```

Comment:
Should we even allow ARRAY_INGEST_MODE as "mvd", i.e. in addition to this, do we need to have a check flagging the queries with CTX_ARRAY_INGEST_MODE = 'mvd'

Reply:
If a user explicitly sets this flag for compaction, it would be out of a conscious choice. It's the same for all the flags above which may not ideally make sense for compaction but are set for some reason (e.g. finalize), and would end up in some valid output. 
We do already have a warn log for array_ingest_mode  = `mvd` in the controller, so not adding another one here.

Reply:
Finalize still makes some sense, however, with MVD mode with compaction is a sure shot way to shoot oneself in the foot. It still exists for ingestion for some historical reasons - We only ingested MVDs + it is difficult to change the queries once the data is ingested as an MVD. For compaction, it makes little sense to modify a string from array to MVD. 

Anyways, we don't need to block this patch for this discussion and can probably take it in a follow-up if need be.